src/platform/unix/mod.rs

```diff
@@ -873,50 +928,27 @@ fn recv(fd: c_int, blocking_mode: BlockingMode)
     Ok((main_data_buffer, channels, shared_memory_regions))
 }
 
-#[cfg(target_os="android")]
-fn maybe_unlink(_: *const c_char) -> c_int {
-    // Calling `unlink` on a file stored on an sdcard immediately deletes it.
-    // FIXME: use a better temporary directory than the sdcard via the Java APIs
-    // and threading that value into Servo.
-    // https://code.google.com/p/android/issues/detail?id=19017
-    0
-}
-
-#[cfg(not(target_os="android"))]
-unsafe fn maybe_unlink(c: *const c_char) -> c_int {
-    libc::unlink(c)
-}
-
-unsafe fn create_memory_backing_store(length: usize) -> c_int {
-    let string = CString::new(TEMP_FILE_TEMPLATE).unwrap();
-    let string_buffer = strdup(string.as_ptr());
-    let fd = mkstemp(string_buffer);
-    assert!(fd >= 0);
-    assert!(maybe_unlink(string_buffer) == 0);
-    libc::free(string_buffer as *mut c_void);
-    assert!(libc::ftruncate(fd, length as off_t) == 0);
-    fd
+#[cfg(any(feature="force-posix-shm", not(target_os="linux")))]
+fn create_shmem(name: CString, length: usize) -> c_int {
+    unsafe {
+        let fd = libc::shm_open(name.as_ptr(),
+                                libc::O_CREAT | libc::O_RDWR | libc::O_EXCL,
+                                0o600);
```

Comment:
So the FreeBSD man page says `shm_unlink` needs write privileges, but testing is showing it needs read-write.